walk-in-interview/src/main/webapp/job-details/script.js

```diff
@@ -7,47 +13,192 @@ const CurrentLocale = 'en';
  */
 import {AppStrings} from '../strings.en.js';
 
-import {JOB_ID_PARAM, setCookie, setErrorMessage} from '../common-functions.js';
+import {JOB_ID_PARAM, setErrorMessage,
+  getRequirementsList} from '../common-functions.js';
+
+import {API} from '../apis.js';
 
-const STRINGS = AppStrings['job-details'];
-const COMMON_STRINGS = AppStrings['common'];
 const UPDATE_JOB_PATH = '../update-job/index.html';
+const HOMEPAGE_PATH = '../index.html';
+
+const COMMON_STRINGS = AppStrings['common'];
+const STRINGS = AppStrings['job-details'];
+const JOB_STRINGS = AppStrings['job'];
+
+window.onload = () => {
+  renderJobDetailsPageElements();
+};
+
+/**
+ * Add the current job (given the jobId) details to the page,
+ * as well as the other elements on the page.
+ */
+async function renderJobDetailsPageElements() {
+  const homepageButton = document.getElementById('back-to-homepage');
+  homepageButton.innerText = STRINGS['back-to-homepage'];
+  homepageButton.addEventListener('click', (_) => {
+    window.location.href= HOMEPAGE_PATH;
+  });
+
+  const jobId = getJobId();
+  if (jobId === '') {
+    setErrorMessage('error-message', /* msg= */ STRINGS['error-message'],
+        /* includesDefaultMsg= */ false);
+    throw new Error('jobId should not be empty');
+  }
+
+  const updateForm = document.getElementById('update-form');
+  updateForm.method = 'GET';
+  updateForm.action = UPDATE_JOB_PATH;
+
+  const jobIdElement = document.getElementById('job-id');
+  jobIdElement.setAttribute('type', 'hidden');
+  jobIdElement.setAttribute('name', JOB_ID_PARAM);
+  jobIdElement.setAttribute('value', jobId);
+
+  const updateButton = document.getElementById('update');
+  updateButton.setAttribute('value', STRINGS['update']);
+  updateButton.setAttribute('type', 'submit');
+
+  const deleteButtonElement = document.getElementById('delete');
+  deleteButtonElement.innerText = STRINGS['delete'];
+
+  const job = await getJobDetails(jobId)
```

Comment:
sync is fine; but not sure if we can consider async? so that we can render page of other part when loading the job details; just an alternative